### ISSUE TITLE
[Feat]: Mock collector 

### ISSUE THREAD

**Issue by shyamvalsan** [netdata/netdata#14051]
### Problem

Maintaining demo spaces for every collector is expensive (especially since some of the paid services only offer short trial periods). We need a way to run the service once, collect real data and then replay it using a mock collector.

### Description

Create a mock collector that can spoof any existing collector

- While developing the collector we test it out with the real service
- Collect raw metric data for 7 days
- Mock collector takes the collected data and replays it endlessly (repeating the same data after 7 day period elapses)

This will save $$ compared to actually hosting each service and having so many different nodes running for purely demo purposes.

### Importance

must have

### Value proposition

1. Saves Netdata money and time
2. Reduces SRE effort to maintain demo space

**Comment by andrewm4894** [comment]
I think we could maybe generalized this to a notion of a "Replay" collector (which i think is what this is maybe we just try and implement it in most useful and geenral way possible) that just replays data from some saved dataset you point it at. 

Such a thing could actually also be very useful for additional testing of ML code when we make changes etc where we could replay various golden datasets on a few test nodes and measure in what ways changes in anomaly rate behaviour is different then expected or existing behaviour etc.

Could imagine some cases where such a thing could even be useful for the community too potentially.

**Comment by shyamvalsan** [comment]
Yes, we could have useful test datasets for different anomaly patterns for example, this might help us validate algorithm or design changes.

**Comment by shyamvalsan** [comment]
Closing in favor of a proper "Netdata Infra Simulator"
